File: backend/app/schemas/isochrone.py
"""Isochrone API response schemas."""

from __future__ import annotations

from pydantic import BaseModel, Field


class IsochroneContour(BaseModel):
    """A single isochrone contour (GeoJSON polygon)."""

    minutes: int
    profile: str
    geometry: dict = Field(default_factory=dict, description="GeoJSON Polygon geometry")


class IsochroneResponse(BaseModel):
    """Response for isochrone endpoint."""

    building_id: str
    latitude: float
    longitude: float
    profile: str = "walking"
    contours: list[IsochroneContour] = []
    mobility_score: float | None = Field(None, ge=0, le=10, description="Mobility score 0-10")
    cached: bool = False
    error: str | None = None

File: frontend/src/api/isochrone.ts
import { apiClient } from '@/api/client';

export interface IsochroneContour {
  minutes: number;
  profile: string;
  geometry: GeoJSON.Polygon;
}

export interface IsochroneResponse {
  building_id: string;
  latitude: number;
  longitude: number;
  profile: string;
  contours: IsochroneContour[];
  mobility_score: number | null;
  cached: boolean;
  error: string | null;
}

export const isochroneApi = {
  get: async (buildingId: string, profile = 'walking', rangeList = '5,10,15'): Promise<IsochroneResponse> => {
    const response = await apiClient.get<IsochroneResponse>(`/buildings/${buildingId}/isochrone`, {
      params: { profile, range_list: rangeList },
    });
    return response.data;
  },
};
